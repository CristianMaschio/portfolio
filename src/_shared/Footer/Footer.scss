@import "../../styles/colors";

footer {
  background-color: $dark-color;
  margin-left: 18rem;
}

footer p {
  margin: 0px;
  padding: 1.5em;
  color: white;
}
