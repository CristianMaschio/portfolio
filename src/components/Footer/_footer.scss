footer {
  background-color: transparent;
  display: flex;
  flex-direction: column;
  height: auto;
  color: white;

  ul {
    flex: 1;
    display: flex;
    align-items: center;
    flex-direction: column;
    list-style-type: none;
    padding: 0;
    margin-top: -40px;
    padding-bottom: 10px;

    @media screen and (max-width: $mobile-max-width) {
      margin-top: -10px;
    }
    li {
      margin: 3px;
      color: white;
      font-size: 10px;
      text-align: center;
      margin-left: 1rem;
      margin-right: 1rem;
      a {
        color: white;
        font-size: 12px;
        text-decoration: none;
      }
    }
  }

  .waves-top-down {
    transform: rotate(180deg);
  }
}
