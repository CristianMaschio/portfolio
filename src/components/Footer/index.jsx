import React from "react";

export default function Footer() {
  return (
    <footer>
      <div className="waves waves-top-down">
        <svg
          fill="white"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 1000 100"
          preserveAspectRatio="none"
        >
          <path d="M 0 14 s 88.64 3.48 300 36 c 260 40 514 27 703 -10 l 12 28 l 3 36 h -1018 z"></path>
          <path d="M 0 45 s 271 45.13 500 32 c 157 -9 330 -47 515 -63 v 86 h -1015 z"></path>
          <path d="M 0 58 s 188.29 32 508 32 c 290 0 494 -35 494 -35 v 45 h -1002 z"></path>
        </svg>
      </div>
      <ul>
        <li>
          <a
            href="mailto:cristianmaschio@gmail.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            CristianMaschio@gmail.com
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/cristian-maschio-37004a146"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://www.linkedin.com/in/cristian-maschio-37004a146
          </a>
        </li>
        <li>
          <a
            href="https://github.com/CristianMaschio"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://github.com/CristianMaschio
          </a>
        </li>
      </ul>
    </footer>
  );
}
