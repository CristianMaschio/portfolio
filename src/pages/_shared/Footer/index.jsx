import React from "react"

export default function Footer() {
  return (
    <footer>
      <ul>
        <li>
          <a
            href="mailto:cristianmaschio@hotmail.it"
            target="_blank"
            rel="noopener noreferrer"
          >
            CristianMaschio@hotmail.it
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/cristian-maschio-37004a146"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://www.linkedin.com/in/cristian-maschio-37004a146
          </a>
        </li>
        <li>
          <a
            href="https://github.com/CristianMaschio"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://github.com/CristianMaschio
          </a>
        </li>
      </ul>
    </footer>
  )
}

