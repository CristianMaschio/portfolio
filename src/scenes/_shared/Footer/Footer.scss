@import "../../../styles/colors";

footer {
  background-color: $dark-color;
  margin-left: 18rem;
  display: flex;
  height: auto;
  color: white;
  ul {
    flex: 1;
    display: flex;
    align-items: center;
    flex-direction: column;
    list-style-type: none;
    li {
      margin: 3px;
      color: white;
      font-size: 10px;
      text-align: center;
      margin-left: 1rem;
      margin-right: 1rem;
      a {
        color: white;
        font-size: 12px;
        text-decoration: none;
      }
    }
    @media screen and (max-width: $mobile-max-width) {
      padding: 0;
    }
  }

  @media screen and (max-width: $mobile-max-width) {
    margin-left: 0px;
    padding: 0;
  }
}
