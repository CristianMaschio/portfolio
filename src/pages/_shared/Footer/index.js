import React, { PureComponent } from "react"
import "./Footer.scss"

export default class Footer extends PureComponent {
  render() {
    return (
      <footer>
        <ul>
          <li>
            <a href="tel:345-273-6368">+39 3452736368</a>
          </li>
          <li>
            <a href="mailto:cristianmaschio@hotmail.it">
              CristianMaschio@hotmail.it
            </a>
          </li>
          <li>
            <a href="https://www.linkedin.com/in/cristian-maschio-37004a146">
              https://www.linkedin.com/in/cristian-maschio-37004a146
            </a>
          </li>
          <li>
            <a href="https://github.com/CristianMaschio">
              https://github.com/CristianMaschio
            </a>
          </li>
        </ul>
      </footer>
    )
  }
}
