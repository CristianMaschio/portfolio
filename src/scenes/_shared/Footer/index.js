import React, { useContext } from "react"
import "./Footer.scss"

import AppState from "../../../gState/context"

export default function Footer() {
  const { language } = useContext(AppState)
  const privacyTerms = require(`../../../data/${language}-content.json`)
    .privacyTerms

  return (
    <footer>
      <ul>
        <li>
          <a href="tel:345-273-6368" target="_blank" rel="noopener noreferrer">
            +39 3452736368
          </a>
        </li>
        <li>
          <a
            href="mailto:cristianmaschio@hotmail.it"
            target="_blank"
            rel="noopener noreferrer"
          >
            CristianMaschio@hotmail.it
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/cristian-maschio-37004a146"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://www.linkedin.com/in/cristian-maschio-37004a146
          </a>
        </li>
        <li>
          <a
            href="https://github.com/CristianMaschio"
            target="_blank"
            rel="noopener noreferrer"
          >
            https://github.com/CristianMaschio
          </a>
        </li>
        <li>{privacyTerms}</li>
      </ul>
    </footer>
  )
}
