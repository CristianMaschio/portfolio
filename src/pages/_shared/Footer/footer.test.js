import React from "react"
import renderer from "react-test-renderer"
import Footer from "."

it("Footer: render correctly", () => {
  const footer = renderer.create(<Footer />).toJSON()
  expect(footer).toMatchSnapshot()
})
