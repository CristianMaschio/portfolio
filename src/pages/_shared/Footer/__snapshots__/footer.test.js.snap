// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer: render correctly 1`] = `
<footer>
  <ul>
    <li>
      <a
        href="mailto:cristianmaschio@hotmail.it"
        rel="noopener noreferrer"
        target="_blank"
      >
        CristianMaschio@hotmail.it
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/cristian-maschio-37004a146"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://www.linkedin.com/in/cristian-maschio-37004a146
      </a>
    </li>
    <li>
      <a
        href="https://github.com/CristianMaschio"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://github.com/CristianMaschio
      </a>
    </li>
  </ul>
</footer>
`;
